data/src/test/java/org/apache/iceberg/io/TestBaseTaskWriter.java

```diff
@@ -0,0 +1,255 @@
+
+  private OutputFileFactory fileFactory = null;
+  private FileAppenderFactory<Record> appenderFactory = null;
+
+  @Parameterized.Parameters(name = "FileFormat = {0}")
+  public static Object[][] parameters() {
+    return new Object[][] {
+        {"avro"},
+        {"parquet"}
+    };
+  }
+
+  public TestBaseTaskWriter(String fileFormat) {
+    super(FORMAT_V2);
+    this.format = FileFormat.valueOf(fileFormat.toUpperCase(Locale.ENGLISH));
+  }
+
+  @Before
+  public void setupTable() throws IOException {
+    this.tableDir = temp.newFolder();
+    Assert.assertTrue(tableDir.delete()); // created by table create
+
+    this.metadataDir = new File(tableDir, "metadata");
+
+    this.table = create(SCHEMA, PartitionSpec.unpartitioned());
+    this.fileFactory = new OutputFileFactory(table.spec(), format, table.locationProvider(), table.io(),
+        table.encryption(), 1, 1);
+
+    int firstFieldId = table.schema().findField("id").fieldId();
+    int secondFieldId = table.schema().findField("data").fieldId();
+    this.appenderFactory = new GenericAppenderFactory(table.schema(), table.spec(),
+        new int[] {firstFieldId, secondFieldId}, table.schema(), null);
+
+    table.updateProperties()
+        .defaultFormat(format)
+        .commit();
+  }
+
+  private Record createRecord(Integer id, String data) {
+    return gRecord.copy("id", id, "data", data);
+  }
+
+  @Test
+  public void testWriteZeroRecord() throws IOException {
+    try (TestTaskWriter writer = createTaskWriter(128 * 1024 * 1024)) {
+      writer.close();
+
+      WriteResult result = writer.complete();
+      Assert.assertEquals(0, result.dataFiles().length);
+      Assert.assertEquals(0, result.deleteFiles().length);
+
+      writer.close();
+      result = writer.complete();
+      Assert.assertEquals(0, result.dataFiles().length);
+      Assert.assertEquals(0, result.deleteFiles().length);
+    }
+  }
+
+  @Test
+  public void testAbort() throws IOException {
+    List<Record> records = Lists.newArrayList();
+    for (int i = 0; i < 2000; i++) {
+      records.add(createRecord(i, "aaa"));
+    }
+
+    WriteResult result;
+    try (TestTaskWriter taskWriter = createTaskWriter(4)) {
+      for (Record record : records) {
+        taskWriter.write(record);
+        taskWriter.delete(record);
+      }
+
+      // Close the current opened files.
+      taskWriter.close();
+
+      // Assert the current data file count.
+      List<Path> files = Files.list(Paths.get(tableDir.getPath(), "data"))
+          .filter(p -> !p.toString().endsWith(".crc"))
+          .collect(Collectors.toList());
+      Assert.assertEquals("Should have 4 files but the files are: " + files, 4, files.size());
+
+      // Abort to clean all delete files and data files.
+      taskWriter.abort();
+
+      // Complete again to get all results.
+      result = taskWriter.complete();
+    }
+    Assert.assertEquals(2, result.deleteFiles().length);
```

Comment:
Just curious, is there a use case for wanting data/delete files after aborting the writer?

Reply:
In real user case, there should be no one calling `complete`  after `abort`.  Here I just want to call `complete` to  get all the delete and data files (The `complete` is the only API to get those files), for asserting purpose.

Reply:
I guess my question would be if we actually want to leave this behavior as is, since it seems like gathering the files after the process getting aborted and files deleted would be not super useful except for testing?

Reply:
I think I agree with Yan here. The data file count assertion above ensures that there are only 4 files. So when the writer aborts, we can use the same logic to check that there are now 0 files instead of asserting that each file individually doesn't exist (below). That would be better because it doesn't seem correct for `result.dataFiles()` to return anything after an abort.

Reply:
OK,  I got your points.  I agree It's better to use the `files` to check the existence after abort. The behavior about getting completed files after abort is not expected.

Thanks @yyanyy and @rdblue .